modules/userId/index.js

```diff
@@ -465,13 +465,109 @@ function getUserIdsAsEids() {
   return createEidsArray(getCombinedSubmoduleIds(initializedSubmodules));
 }
 
+/**
+* This function will be exposed in the global-name-space so that userIds can be refreshed after initialization.
+* @param {String[]} submoduleNames submodules to be refreshed
+*/
+function refreshUserIds(submoduleNames) {
```

Comment:
Would we want `submoduleNames` to be optional, with it defaulting to refreshing all submodules? I brought up passing in the submodule names because my particular use case only requires refreshing one or two userId modules, but refreshing all might still be a useful default? If we do end up making `submoduleNames` optional, I would suggest making the argument to `refreshUserIds` an object so it would be easy to pass additional arguments down the line in a non-breaking way.

Note this is mostly a question. It may be the case we want to keep it the way it is with submoduleNames being required.

Reply:
not a bad idea to support refreshing all ids. pubs may not know which of their ids need a refresh per se, so being able to just refresh them all seems like an easy way to do it for them.

maybe in the call to `getId` it should be flagged if a force refresh is happening in case the module wants to act differently?